client/components/data/query-whois/index.jsx

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingWhois } from 'state/selectors';
+import { requestWhois } from 'state/domains/management/actions';
+
+class QueryWhois extends Component {
+	componentWillMount() {
+		if ( this.props.requesting ) {
+			return;
+		}
+		this.props.requestWhois( this.props.domain );
+	}
+
+	render() {
+		return null;
+	}
+}
+
+QueryWhois.propTypes = {
+	domain: PropTypes.string,
+	requesting: PropTypes.bool,
+	requestWhois: PropTypes.func
+};
+
+export default connect(
+	( state, { domain } ) => ( { requesting: isRequestingWhois( state, domain ) } ),
```

Comment:
Is it enough to just move the creation out of the connect call, like this?

```
const mapStateToProps = ( state, { domain } ) =>
	( { requesting: isRequestingWhois( state, domain ) } );

export default connect(
	mapStateToProps,
	{ requestWhois }
)( QueryWhois );
```
I think that guarantees that the function is created only once in the module (with `this` bound to the module, but we don't really care about that aspect).